pkg/controller/master-controller-manager/kubeone/controller.go

```diff
@@ -267,79 +273,83 @@ func (r *reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, exte
 }
 
 func (r *reconciler) initiateImportAction(ctx context.Context, log *zap.SugaredLogger, externalCluster *kubermaticv1.ExternalCluster) error {
-	kubeconfigSecret := &corev1.Secret{}
-	if err := r.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: kubernetesprovider.GetKubeOneNamespaceName(externalCluster.Name), Name: resources.KubeOneKubeconfigSecretName}, kubeconfigSecret); err != nil {
-		if apierrors.IsNotFound(err) {
-			if err := r.importCluster(ctx, log, externalCluster); err != nil {
-				log.Debug("failed to import kubeone cluster %w", err)
-				return err
-			}
-			// update kubeone externalcluster status.
-			if err := r.updateClusterStatus(ctx, externalCluster, kubermaticv1.ExternalClusterCondition{
-				Phase: kubermaticv1.ExternalClusterPhaseRunning,
-			}); err != nil {
-				return err
-			}
+	if externalCluster.Spec.KubeconfigReference == nil {
+		externalCluster, err := r.importCluster(ctx, log, externalCluster)
+		if err != nil {
+			log.Debug("failed to import kubeone cluster, ", err)
```

Comment:
Why is this printed and why on the debug level? Is the err returned by this function ignored?

Also if anything it should be `log.Debugw("fialed to ....", zap.Error(err))`

Reply:
yes, we are not returning the error just logging it, coz we dont want retries,  mostly if there is an error in import it will be there in next retry too

Reply:
i will make the change to `log.Debugw("fialed to ....", zap.Error(err))`